src/gui/transferlistsortmodel.cpp

```diff
@@ -39,17 +39,27 @@
 
 namespace
 {
+    template<typename  T>
+    inline int threeWayCompare(const T& left, const T &right)
+    {
+        return (left > right) ? 1 : ((left == right) ? 0 : -1);
+    }
+
     template <typename T>
-    bool customLessThan(const T left, const T right)
+    int customCompare(const T left, const T right)
     {
         static_assert(std::is_arithmetic_v<T>);
 
         const bool isLeftValid = (left >= 0);
         const bool isRightValid = (right >= 0);
 
         if (isLeftValid && isRightValid)
-            return left < right;
-        return isLeftValid;
+            return (sizeof(T) <= sizeof(int))
+                       ? (left - right)
```

Comment:
I would prefer `customCompare` to return one of three possible values: -1, 0, 1 (i.e. less than, equal to, greater than). What's the point to check type size above if you can apply the same comparison algorithm in all the cases?

Reply:
> What's the point to check type size above if you can apply the same comparison algorithm in all the cases?

I was trying to save one comparison.

Reply:
>I was trying to save two comparisons.

...by perform one extra comparison.

Reply:
> (sizeof(T) <= sizeof(int))

is a compile-time check, it would have removed in compiled code.

Reply:
If it were that simple, I don't think they would have to invent `if constexpr`, etc.

Reply:
even without any optimization enabled both GCC and clang was able to remove this check https://godbolt.org/z/j9bv7e

Reply:
Well, modern compilers are smart enough. The question is how confident we can be about this without looking at the resulting assembler every time.

BTW, if you were sure it was good, why did you give up so easily?

Reply:
>  if you were sure it was good, why did you give up so easily?

I'm 99% sure you guys would have asked to remove it.

Reply:
If this optimization really makes sense, I would put it in `threeWayCompare()` instead.

Reply:
> optimization really makes sense

IMO it shouldn't cause any real performance difference at all.